include/xgboost/common/quantile.h

```diff
@@ -493,34 +522,47 @@ struct WQSummary {
     // Choose entrys.
     RType last_selected_entry_value = std::numeric_limits<RType>::min();
     size_t select_count = 0;
+    Entry lastEntry = items[0].entry;
     for (size_t i = 1; i < items.size(); ++i) {
-      bool do_select = !items[i - 1].has_entry && items[i].has_entry &&
-                       items[i].entry.value != last_selected_entry_value;
+      //   CASE max_index<=maxsize:All unique item will be select
+      //   CASE other : select unique after dx2 index
+      bool do_select = ObliviousChoose(
+          ObliviousLess(max_index , maxsize),
```

Comment:
I think this might need to be `ObliviousLessOrEqual`? (since in your original code it was `<=`, right?)

Reply:
yes， but it won`t effect ,cause = could be processd right in both cases

Reply:
I see. In that case, please update the comment on line 527 (the comment says `<=`)? Just to prevent confusion for someone who reads the code later and thinks it's a bug.